python/ray/tune/trial.py

```diff
@@ -283,6 +283,9 @@ def should_stop(self, result):
         if result.get(DONE):
             return True
 
+        if callable(self.stopping_criterion):
+            return self.stopping_criterion(result)
```

Comment:
can you actually extend this to take in `trial_id` too? so `self.stopping_criteria(trial_id, result)`?